libzim doesn't compile on windows.

See https://ci.appveyor.com/project/Kiwix/kiwix-build/builds/34908813

It seems that there is a ambiguity with the possible implementations of `!=` within the lru_cache.
(Between some implemation in (MSVC) `xmemory` and in [`zim_types.h`](https://github.com/openzim/libzim/blob/master/src/zim_types.h#L65-L66)

```
"cl" "-Isrc\libzim.a.p" "-Isrc" "-I..\src" "-Iinclude" "-I..\include" "-Istatic" "-IC:/extra\include" "-IC:/extra/include" "/MD" "/nologo" "/showIncludes" "/W2" "/EHsc" "/std:c++14" "/permissive-" "/Ob2" "/Gw" "/O2" "-DLZMA_API_STATIC" "-DSORTPP_PASS" "/Fdsrc\libzim.a.p\fileimpl.cpp.pdb" /Fosrc/libzim.a.p/fileimpl.cpp.obj "/c" ../src/fileimpl.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\list(1731): error C2593: 'operator !=' is ambiguous
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\xmemory0(1052): note: could be 'bool std::operator !=<std::_List_node<_Ty,std::_Default_allocator_traits<_Alloc>::void_pointer>,std::_List_node<_Ty,std::_Default_allocator_traits<_Alloc>::void_pointer>>(const std::allocator<std::_List_node<_Ty,std::_Default_allocator_traits<_Alloc>::void_pointer>> &,const std::allocator<std::_List_node<_Ty,std::_Default_allocator_traits<_Alloc>::void_pointer>> &) noexcept'
        with
        [
            _Ty=std::pair<zim::article_index_t,std::shared_ptr<const zim::Dirent>>,
            _Alloc=std::allocator<std::pair<zim::article_index_t,std::shared_ptr<const zim::Dirent>>>
        ]
c:\projects\kiwix-build\libzim\src\zim_types.h(65): note: or       'bool operator !=<std::allocator<std::_List_node<_Ty,std::_Default_allocator_traits<_Alloc>::void_pointer>>>(const T &,const T &)'
        with
        [
            _Ty=std::pair<zim::article_index_t,std::shared_ptr<const zim::Dirent>>,
            _Alloc=std::allocator<std::pair<zim::article_index_t,std::shared_ptr<const zim::Dirent>>>,
            T=std::allocator<std::_List_node<std::pair<zim::article_index_t,std::shared_ptr<const zim::Dirent>>,std::_Default_allocator_traits<std::allocator<std::pair<zim::article_index_t,std::shared_ptr<const zim::Dirent>>>>::void_pointer>>
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\list(1731): note: while trying to match the argument list '(std::allocator<std::_List_node<_Ty,std::_Default_allocator_traits<_Alloc>::void_pointer>>, std::allocator<std::_List_node<_Ty,std::_Default_allocator_traits<_Alloc>::void_pointer>>)'
        with
        [
            _Ty=std::pair<zim::article_index_t,std::shared_ptr<const zim::Dirent>>,
            _Alloc=std::allocator<std::pair<zim::article_index_t,std::shared_ptr<const zim::Dirent>>>
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\list(1675): note: while compiling class template member function 'void std::list<std::pair<key_t,value_t>,std::allocator<_Ty>>::_Splice(std::_List_const_iterator<std::_List_val<std::_List_simple_types<_Ty>>>,std::list<_Ty,std::allocator<_Ty>> &,std::_List_const_iterator<std::_List_val<std::_List_simple_types<_Ty>>>,std::_List_const_iterator<std::_List_val<std::_List_simple_types<_Ty>>>,unsigned __int64)'
        with
        [
            key_t=zim::article_index_t,
            value_t=std::shared_ptr<const zim::Dirent>,
            _Ty=std::pair<zim::article_index_t,std::shared_ptr<const zim::Dirent>>
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\list(1441): note: see reference to function template instantiation 'void std::list<std::pair<key_t,value_t>,std::allocator<_Ty>>::_Splice(std::_List_const_iterator<std::_List_val<std::_List_simple_types<_Ty>>>,std::list<_Ty,std::allocator<_Ty>> &,std::_List_const_iterator<std::_List_val<std::_List_simple_types<_Ty>>>,std::_List_const_iterator<std::_List_val<std::_List_simple_types<_Ty>>>,unsigned __int64)' being compiled
        with
        [
            key_t=zim::article_index_t,
            value_t=std::shared_ptr<const zim::Dirent>,
            _Ty=std::pair<zim::article_index_t,std::shared_ptr<const zim::Dirent>>
        ]
c:\projects\kiwix-build\libzim\src\lrucache.h(51): note: see reference to class template instantiation 'std::list<std::pair<key_t,value_t>,std::allocator<_Ty>>' being compiled
        with
        [
            key_t=zim::article_index_t,
            value_t=std::shared_ptr<const zim::Dirent>,
            _Ty=std::pair<zim::article_index_t,std::shared_ptr<const zim::Dirent>>
        ]

--- a/src/fileimpl.cpp
+++ b/src/fileimpl.cpp
@@ -276,7 +276,7 @@ offset_t readOffset(const Reader& reader, size_t idx)
       throw ZimFileFormatError("article index out of range");
 
     pthread_mutex_lock(&direntCacheLock);
-    auto v = direntCache.get(idx);
+    auto v = direntCache.get(idx.v);
     if (v.hit())
     {
       log_debug("dirent " << idx << " found in cache; hits "
@@ -329,7 +329,7 @@ offset_t readOffset(const Reader& reader, size_t idx)
 
     log_debug("dirent read from " << indexOffset);
     pthread_mutex_lock(&direntCacheLock);
-    direntCache.put(idx, dirent);
+    direntCache.put(idx.v, dirent);
     pthread_mutex_unlock(&direntCacheLock);
 
     return dirent;
@@ -368,7 +368,7 @@ offset_t readOffset(const Reader& reader, size_t idx)
     if (idx >= getCountClusters())
       throw ZimFileFormatError("cluster index out of range");
 
-    return clusterCache.getOrPut(idx, [=](){ return readCluster(idx); });
+    return clusterCache.getOrPut(idx.v, [=](){ return readCluster(idx); });
   }
 
   offset_t FileImpl::getClusterOffset(cluster_index_t idx) const

--- a/src/fileimpl.h
+++ b/src/fileimpl.h
@@ -52,11 +52,11 @@ namespace zim
       std::unique_ptr<const Reader> urlPtrOffsetReader;
       std::unique_ptr<const Reader> clusterOffsetReader;
 
-      lru_cache<article_index_t, std::shared_ptr<const Dirent>> direntCache;
+      lru_cache<article_index_type, std::shared_ptr<const Dirent>> direntCache;
       pthread_mutex_t direntCacheLock;
 
       typedef std::shared_ptr<const Cluster> ClusterHandle;
-      ConcurrentCache<cluster_index_t, ClusterHandle> clusterCache;
+      ConcurrentCache<cluster_index_type, ClusterHandle> clusterCache;
 
       bool cacheUncompressedCluster;
       typedef std::map<char, article_index_t> NamespaceCache;

